apps/condo/domains/common/hooks/usePrevious.tsx

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useRef } from 'react'
+
+/**
+ * Monitoring some variable value
+ * After it's change useEffect will update ref
+ * But before that will be returned previous value
+ * @param value to monitor
+ */
+export function usePrevious<T> (value: T): T {
+    const ref = useRef<T>()
```

Comment:
Why not `useState(value)`?

Reply:
useState unlike useRef changes it's value not instantly. That's why trick with useEffect will not work :)

The trick here is to push value to update in useEffect (async) and return old value, which is not changed by that time. So we can have state where `previous !== current`. After useEffect worked in async we ll have new state where `current === current`